lib/livebook/runtime/epmd.ex

```diff
@@ -47,12 +47,28 @@ defmodule Livebook.Runtime.EPMD do
   def port_please(name, host), do: port_please(name, host, :infinity)
 
   def port_please(name, host, timeout) do
-    case livebook_port(name) do
-      0 -> :erl_epmd.port_please(name, host, timeout)
-      port -> {:port, port, @epmd_dist_version}
+    # If the target node is on the same host, check if it's a Livebook
+    # server or runtime to bypass EPMD. If it is a different node, we
+    # always fall back to :erl_epmd, even if it's a Livebook node.
+    if host() == normalize_host(host) do
```

Comment:
Gah, this doesn't work with domains, the host `argument` will be an IP anyway. I think we need to normalize with `:inet.gethostbyname`.